src/Commands/User.php

```diff
@@ -0,0 +1,693 @@
+                'yellow'
+            );
+        }
+    }
+
+    /**
+     * Remove a user by username or email from a group
+     *
+     * @param string|null $group    Group to remove user from
+     * @param string|null $username User name to search for (optional)
+     * @param string|null $email    User email to search for (optional)
+     */
+    private function removegroup($group = null, $username = null, $email = null): void
+    {
+        if ($group === null) {
+            $group = $this->prompt('Group', null, 'required');
+        }
+
+        $user = $this->findUser('Remove user from group', $username, $email);
+
+        $confirm = $this->prompt(
+            'Remove the user "' . $user->username . '" from the group "' . $group . '" ?',
+            ['y', 'n']
+        );
+
+        if ($confirm === 'y') {
+            $user->removeGroup($group);
+
+            $this->write('User "' . $user->username . '" removed from group "' . $group . '"', 'green');
+        } else {
+            $this->write('Removal of the user "' . $user->username . '" from the group "' . $group . '" cancelled', 'yellow');
+        }
+    }
+
+    /**
+     * Find an existing user by username or email.
+     *
+     * @param string      $question Initial question at user prompt
+     * @param string|null $username User name to search for (optional)
+     * @param string|null $email    User email to search for (optional)
+     */
+    private function findUser($question = '', $username = null, $email = null): UserEntity
+    {
+        if ($username === null && $email === null) {
+            $choice = $this->prompt($question . ' by username or email ?', ['u', 'e']);
+
+            if ($choice === 'u') {
+                $username = $this->prompt('Username', null, 'required');
+            } elseif ($choice === 'e') {
+                $email = $this->prompt('Email', null, 'required|valid_email');
+            }
+        }
+
+        $userModel = model(UserModel::class);
+        $userModel
+            ->select($this->tables['users'] . '.id as id, username, secret')
+            ->join(
+                $this->tables['identities'],
+                $this->tables['users'] . '.id = ' . $this->tables['identities'] . '.user_id',
+                'LEFT'
+            )
+            ->groupStart()
+            ->where($this->tables['identities'] . '.type', Session::ID_TYPE_EMAIL_PASSWORD)
+            ->orGroupStart()
+            ->where($this->tables['identities'] . '.type', null)
+            ->groupEnd()
+            ->groupEnd()
+            ->asArray();
+
+        $user = null;
+        if ($username !== null) {
+            $user = $userModel->where('username', $username)->first();
+        } elseif ($email !== null) {
+            $user = $userModel->where('secret', $email)->first();
+        }
+
+        if ($user === null) {
+            $this->write("User doesn't exist", 'red');
+
+            throw new RuntimeException("User doesn't exist");
```

Comment:
In all cases, I disagree with throw.

Reply:
The exception will be caught in `run()` method.
It is the replacement of `exit()`.